src/plugin.ts

```diff
@@ -199,6 +199,13 @@ class LiftPlugin {
         if (statements.length === 0) {
             return;
         }
+
+        const role = this.serverless.service.provider.iam?.role;
+
+        if (typeof role === "object" && "statements" in role) {
+            role.statements?.push(...statements);
```

Comment:
shouldn't you `return` here as well to prevent creation of `this.serverless.service.provider.iamRoleStatements` in case version of sls is up to date ?

Reply:
indeed, did it in my first commit but forgot it when refactoring to fix the type issue :+1: